src/machine/machinectl.c

```diff
@@ -1446,7 +1445,7 @@ static int login_machine(int argc, char *argv[], void *userdata) {
                          "member='MachineRemoved',"
                          "arg0='", machine, "'");
 
-        r = sd_bus_add_match_async(bus, &slot, match, on_machine_removed, NULL, &forward);
+        r = sd_bus_add_match_async(bus, NULL, match, on_machine_removed, NULL, &forward);
```

Comment:
this doesn't look right... the bus lives longer than this function call, and the bus slot will remain installed hence after returning, with userdata pointing to the forward object, which is not OK...

Slot objects are only redundant if we know the bus is going away at a time where the userdata objects we bus to them are still valid. We can't really allow slots to exist with invalid userdata objects...